### PR TITLE
Add tensorleap-infra chart with elastic operator and use elastic version 7.17 (support auto-migration)

### PR DESCRIPTION
**Pull request by moshe-kabala**
- Add elastic operator into crds directory
- Update elastic search templates - use eck operator Kind and update the pvc name
- Update default elastic url for node and engine
- Change the manifest - add support for `tensorleap-infra` chart and adding installer and app version to enforce CLI upgrade and reinstall
- Refactor the installation process to support reinstall and the new installation manifest version

This PR uses an `elasticsearch` operator to install elastic - it works on a clean environment but needs more testing on environments with DNS problems

The way we chose to upgrade the data is by first upgrading to 7.17 and then to the latest version 8.8.1

This branch also added support in `reinstall` - since it required support for the new elastic PVC

## Tests
Since this branch refactored the whole installation flow to support the new installation manifest and support reinstall on both `install` and `upgrade` in addition to upgrading the elastic, I did the following test to make sure we don't have any regressions
- Run `Install`, `Upgrade` `Reinstall` commands - and then import projects and train 
- Install older version import all the projects(7) with old CLI and then run CLI upgrade and make sure the elastic migration worked well, then try to train projects
- Test `create-manifest` and `pack-installation` commands
- Run airgap installation
- Test the reinstall functionality (app version change or installation params changed)

### CODE REVIEW

File: .github/workflows/update_images.yml [hunk 1]
```diff
@@ -89,9 +89,11 @@ jobs:
           helm repo add minio https://charts.min.io
           helm dependency build ./charts/tensorleap
           rm ./charts/tensorleap/Chart.lock
-          helm template ./charts/tensorleap \
+          helm dependency build ./charts/tensorleap-infra
+          rm ./charts/tensorleap-infra/Chart.lock
```
Comment (yotamazriel) [inline .github/workflows/update_images.yml:93]: I'm not sure we want to remove the lock file

**Review by yotamazriel — APPROVED** [review]
Review my comments
@asafyehezkel Let's sync and add them to pivotal tracker
Nothing holding us from merging

File: pkg/server/manifest/manifest.go
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/server/manifest/manifest.go:1]: Currently I move the installer version to detected package called version

File: pkg/server/installation_params.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+			&map[string]interface{}{"clusterPort": installationParams.ClusterPort, "error": err.Error()})
+		return nil, err
+	}
+
+	if err := InitRegistryPort(&installationParams.RegistryPort); err != nil {
+		log.SendCloudReport("error", "Failed initializing registry port", "Failed",
+			&map[string]interface{}{"registryPort": installationParams.RegistryPort, "error": err.Error()})
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/server/installation_params.go:99]: done

File: .github/workflows/update_images.yml (hunk 1 above)
Comment (moshe-kabala) [reply .github/workflows/update_images.yml:93]: I will keep it until we will check if it have any influence on chart releases

**Pull request merged**